postgresql-ballerina/Module.md

```diff
@@ -0,0 +1,380 @@
+
+This example demonstrates querying data from a table in a database. 
+First, a type is created to represent the returned result set. Note the mapping of the database column 
+to the returned record's property is case-insensitive (i.e., the `ID` column in the result can be mapped to the `id` 
+property in the record). Next, the `SELECT` query is executed via the `query` remote function of the client. 
+Once the query is executed, each data record can be retrieved by looping the result set. The `stream`  
+returned by the `SELECT` operation holds a pointer to the actual data in the database and it loads data from the table 
+only when it is accessed. This stream can be iterated only once. 
+
+```ballerina
+// Define a type to represent the results.
+type Student record {
+    int id;
+    int age;
+    string name;
+};
+
+// Select the data from the database table. The query parameters are passed 
+// directly. Similar to the `execute` examples, parameters can be passed as
+// sub types of `sql:TypedValue` as well.
+int id = 10;
+int age = 12;
+sql:ParameterizedQuery query = `SELECT * FROM students
+                                WHERE id < ${id} AND age > ${age}`;
+stream<Student, sql:Error> resultStream = dbClient->query(query);
+
+// Iterating the returned table.
+error? e = resultStream.forEach(function(Student student) {
+   //Can perform any operations using 'student' and can access any fields in the returned record of type `Student`.
+});
+```
+
+Defining the return type is optional. Hence, the above example can be modified as follows with an open record 
+type as the return type. The field name in the open record type will be the same as how the column is defined 
+in the database.
+
+```ballerina
+// Select the data from the database table. The query parameters are passed 
+// directly. Similar to the `execute` examples, parameters can be passed as 
+// sub types of `sql:TypedValue` as well.
+int id = 10;
+int age = 12;
+sql:ParameterizedQuery query = `SELECT * FROM students
+                                WHERE id < ${id} AND age > ${age}`;
+stream<record{}, sql:Error> resultStream = dbClient->query(query);
+
+// Iterating the returned table.
+error? e = resultStream.forEach(function(record{} student) {
+    //Can perform any operations using 'student' and can access any fields in the returned record.
+});
+```
+
+There are situations in which you may not want to iterate through the database and in that case, you may decide
+to only use the `next()` operation in the result `stream` and retrieve the first record. In such cases, the returned
+result stream will not be closed and you have to invoke the `close` operation explicitly on the 
+`sql:Client` to release the connection resources and avoid a connection leak as shown below.
+
+```ballerina
+stream<Student, sql:Error> resultStream = dbClient->query("SELECT count(*) as total FROM students");
+record {|Student value;|}|error? result = resultStream.next();
```

Comment:
```suggestion
Student|error? result = resultStream.next();
```

Reply:
This should work

Reply:
We can't return record data directly as stream's next() returns a record with the next member under the value field if not empty or its completion.

Reply:
Can we something like this

```
Student? result = check resultStream.next().value;
if value is Student {        
    // An error is returned as the result.    
} else { 
   // The `Student` table must be empty.    
}
```